Formula/go.rb

```diff
@@ -38,8 +38,8 @@ class Go < Formula
   # Don't update this unless this version cannot bootstrap the new version.
   resource "gobootstrap" do
     on_macos do
-      url "https://storage.googleapis.com/golang/go1.7.darwin-amd64.tar.gz"
-      sha256 "51d905e0b43b3d0ed41aaf23e19001ab4bc3f96c3ca134b48f7892485fc52961"
+      url "https://golang.org/dl/go1.12.17.darwin-amd64.tar.gz"
+      sha256 "a2f58b4bf10f3e882c1a43eba27a2aba65fc815384fbdfc46331c13bca5f5f24"
```

Comment:
I have no objection to bootstrapping with a recent compiler. We'll probably need a recent version for ARM anyway

Reply:
I haven't tried to build an ARM go with this compiler. I can get back to you with the lowest version that compiles the patched Go trunk later!

Reply:
I can confirm that go1.12.17 can build a native arm go on macOS 11.1

```
 ~/D/g/src git:() ➜ arch --x86_64 env GODEBUG=asyncpreemptoff=1 GOOS=darwin GOARCH=arm64 ./make.bash 
Building Go cmd/dist using /Users/arnaud/go1.4. (go1.12.17 darwin/amd64)
Building Go toolchain1 using /Users/arnaud/go1.4.
Building Go bootstrap cmd/go (go_bootstrap) using Go toolchain1.
Building Go toolchain2 using go_bootstrap and Go toolchain1.
Building Go toolchain3 using go_bootstrap and Go toolchain2.
Building packages and commands for host, darwin/amd64.
Building packages and commands for target, darwin/arm64.
---
Installed Go for darwin/arm64 in /Users/arnaud/Desktop/goarm
Installed commands in /Users/arnaud/Desktop/goarm/bin
 ~/D/g/src git:() ➜ ../bin/go version
go version devel +7716a2fbb7 Sun Nov 22 11:44:49 2020 -0500 darwin/amd64
```

Reply:
I can confirm that 1.12.17 can build a native arm64 go.

 ``` ~/D/g/src git:() ➜ arch --x86_64 env GODEBUG=asyncpreemptoff=1 GOOS=darwin GOARCH=arm64 ./make.bash 
Building Go cmd/dist using /Users/arnaud/go1.4. (go1.12.17 darwin/amd64)
Building Go toolchain1 using /Users/arnaud/go1.4.
Building Go bootstrap cmd/go (go_bootstrap) using Go toolchain1.
Building Go toolchain2 using go_bootstrap and Go toolchain1.
Building Go toolchain3 using go_bootstrap and Go toolchain2.
Building packages and commands for host, darwin/amd64.
Building packages and commands for target, darwin/arm64.
---
Installed Go for darwin/arm64 in /Users/arnaud/Desktop/goarm
Installed commands in /Users/arnaud/Desktop/goarm/bin

 ~/D/g/src git:() ➜ ../bin/darwin_arm64/go version
go version devel +7716a2fbb7 Sun Nov 22 11:44:49 2020 -0500 darwin/arm64
```

bin/go is a amd64 build for some reason, but it's probably something that should wait until 1.16 is released, they might change some stuff.

Reply:
> bin/go is a amd64 build for some reason

Then it's not a native arm64 go, is it? I don't understand that contradiction.

Reply:
I failed to mention this in my original PR, but this PR itself isn't sufficient
Which is why it's probably best if it's not merged until 1.16 (but this can be used as a reference of what needs to be done). I wanted to explore why bootstraped Go was crashing (as reported in https://github.com/Homebrew/brew/issues/7857), and prepare some of the investigative work needed for 1.16.

>Then it's not a native arm64 go, is it? I don't understand that contradiction.

I don't get that either, but if you build with [this command](https://github.com/golang/go/issues/42684#issuecomment-731836687), you get:
- amd64 go in bin/go
- arm64 go in bin/darwin_arm64/go

bin/darwin_arm64/ contains a fully working distribution that works perfectly if you move it somewhere and set it as your GOROOT: I've been using that for a couple days now.
Not sure why an amd go gets built in bin/ but I'm not really familiar with Go's make.bash and bootstrap.bash. It may be a simple build script difference.